pkg/ccl/backupccl/backup_processor.go

```diff
@@ -22,16 +23,32 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
 	"github.com/cockroachdb/cockroach/pkg/sql/rowexec"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/storage/cloud"
 	"github.com/cockroachdb/cockroach/pkg/storage/cloudimpl"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
-	hlc "github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 	"github.com/cockroachdb/errors"
 	gogotypes "github.com/gogo/protobuf/types"
 )
 
+// writeInProcessor controls whether we write the SST files of the backup
+// from the SQL processor (rather than from the KV layer). Tenant backups always
+// write the file at the processor level.
+//
+// We want to be able to control when this writing happens since it is beneficial to
+// do it at the processor level when inside a multi-tenant cluster since we can control
+// connections to external resources on a per-tenant level rather than at the shared
+// KV level. It also enables tenant access to `userfile` destinations, which store the
+// date in SQL. However, this does come at a cost since we need to incur an extra copy
+// of the data and therefore increase network and memory utilization in the cluster.
+var writeInProcessor = settings.RegisterBoolSetting(
```

Comment:
I'm a bit lukewarm on the setting -- I'm not sure when we'd ever want to flip it (outside the test that I might just do by testing a tenant instead)

Reply:
Agreed -- it was mainly here for testing, but I like rebasing on #57352 and actually testing inside a tenant.

Reply:
Now that  #57352  is in, I think this patch could include unskipping the skipped test it added